gems/sorbet/lib/constant_cache.rb

```diff
@@ -103,6 +103,8 @@ def name_by_class(klass)
     name = Sorbet::Private::RealStdlib.real_name(mod)
     Sorbet::Private::Status.say("Naming #{name}", print_without_tty: false)
     return if name == 'RSpec::ExampleGroups' # These are all anonymous classes and will be quadratic in the number of classes to name them. We also know they don't have any hidden definitions
+    # JRuby crashes on access to some constants on some Java proxy objects, so exclude all of them.
+    return if name == 'Java'
```

Comment:
Do you only want to skip `::Java` or also `Foo::Java::Bar`? If the former I think adding an `owner` check of some sort would work.

Reply:
+1, could we be more specific in this test? 
Having any class named `Java` ignored by cache would be surprising.

Reply:
> +1, could we be more specific in this check?

I've opened an issue with the JRuby project (jruby/jruby#5783) and their recommendation is to check for the two constants which are problematic, `NEVER` and `UNDEF`. A specific example that caused this bug for me is `Java::ComHeadiusRacc::Cparse::CparseParams::NEVER`.

I think we could check for constants in modules under `::Java`. I was thinking of building a `blacklisted?(constant)` wrapper around `ConstantCache::DEPRECATED_CONSTANTS` that also checks for constants that start with `Java` and end with `NEVER`/`UNDEF`. Does that sound like a good idea to you do you have a better approach in mind?

Reply:
>  their recommendation is to check for the two constants which are problematic, NEVER and UNDEF

Could we check for those specific names?

> Java::ComHeadiusRacc::Cparse::CparseParams::NEVER
and, I guess
> Java::ComHeadiusRacc::Cparse::CparseParams:: UNDEF
?
or are there more?

Reply:
The ones I've found are:

- `Java::ComHeadiusRacc::Cparse::CparseParams::NEVER`
- `Java::ComHeadiusRacc::Cparse::CparseParams::UNDEF`
- `Java::ComHeadiusRacc::Cparse::Parser::NEVER`
- `Java::ComHeadiusRacc::Cparse::Parser::UNDEF`
- `Java::OrgJruby::RubyBasicObject::NEVER`
- `Java::OrgJruby::RubyBasicObject::UNDEF`
- `Java::OrgJruby::RubyClass::NEVER`
- `Java::OrgJruby::RubyClass::UNDEF`
- `Java::OrgJruby::RubyModule::NEVER`
- `Java::OrgJruby::RubyModule::UNDEF`
- `Java::OrgJruby::RubyObject::NEVER`
- `Java::OrgJruby::RubyObject::UNDEF`

The next release of racc should fix the ones in `ComHeadiusRacc`. I don't think there's much that can be done about the ones in `OrgJruby`. However, from the JRuby issue:

> most ext loading should be safe but we might found a library that isn't updated such as you did.

So there might be other libraries out there that have this issue. The easiest way forward would probably be to add the constants above to `DEPRECATED_CONSTANTS` and hope that not too many more show up in the wild?

Reply:
> The easiest way forward would probably be to add the constants above to DEPRECATED_CONSTANTS and hope that not too many more show up in the wild?

I'd be onboard with adding them as they are discovered.
Would you kindly take a stab on this? Thank you!